public/model/selected-admins.js

```diff
@@ -58,6 +63,26 @@ var SelectedAdmins = P({
     );
   },
 
+  search_admin: function(admin_code, on_unsearch) {
+    if (this.is_admin_searched(admin_code)) {
+      console.log('Admin', admin_code, 'already selected, not doing anything.');
+      return;
+    }
+
+    // We only call the previously searched admins' on_unsearch callbacks after
+    // updating `searched_admin_codes`. This is so that when the callbacks run,
+    // `searched_admin_codes` accurately reflects what is now searched.
+    var unsearch_cbs = _.values(this.searched_admin_codes);
+    this.searched_admin_codes = {};
+    this.searched_admin_codes[admin_code] = on_unsearch || _.noop;
+    unsearch_cbs.forEach(function(cb) { cb(); });
+    this.selection_ee.emit(
+      new SelectionEvents.AdminSearchEvent(this.get_admin_codes())
+    );
+
+    this.onUpdate();
```

Comment:
we're not doing `onUpdate()` anymore, just make the AdminSearchEvent handler in `main.jsx` call `rerender`.